File: dongyimai_sellergoods_interface/src/main/java/com/offcn/sellergoods/service/GoodsService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.group.Goods;
import com.offcn.pojo.TbGoods;
import com.offcn.pojo.TbItem;

import java.util.List;

public interface GoodsService {

	//findAll
	public List<TbGoods> findAll();

	//findPage
	public PageResult findPage(int pageNum, int pageSize);

	//add（已修改，需求：新增保存商品SPU表的同时，保存对应的扩展表、SKU表）（复合实体类Goods）
	public void add(Goods goods);
	
	//update（已修改，需求：修改保存商品SPU表的同时，保存对应的扩展表、SKU表）（复合实体类Goods）
	public void update(Goods goods);
	
	//findOne（已修改，需求：根据商品ID，查询SPU信息、附带扩展表信息、SKU表信息）（复合实体类Goods）
	public Goods findOne(Long id);

	//delete
	public void delete(Long[] ids);

	//条件、模糊查询+分页
	public PageResult findPage(TbGoods goods, int pageNum, int pageSize);

	/**
	 * 批量审核商品
	 * @param ids
	 * @param status
	 */
	public void updateStatus(Long[] ids,String status);

	/**
	 * 通过商品ID和状态查询SKU集合
	 * @param ids
	 * @param status
	 * @return
	 */
	public List<TbItem> findItemListByGoodsIdsAndStatus(Long[] ids, String status);
	
}

File: dongyimai_manager_web/src/main/webapp/js/service/goodsService.js
//服务层
app.service('goodsService',function($http){
	    	
	//findAll
	this.findAll=function(){
		return $http.get('../goods/findAll.do');		
	}

	//findPage
	this.findPage=function(page,rows){
		return $http.get('../goods/findPage.do?page='+page+'&rows='+rows);
	}

	//findOne
	this.findOne=function(id){
		return $http.get('../goods/findOne.do?id='+id);
	}

	//add
	this.add=function(entity){
		return  $http.post('../goods/add.do',entity );
	}

	//update
	this.update=function(entity){
		return  $http.post('../goods/update.do',entity );
	}

	//dele
	this.dele=function(ids){
		return $http.get('../goods/delete.do?ids='+ids);
	}

	//条件、模糊查询+分页
	this.search=function(page,rows,searchEntity){
		return $http.post('../goods/search.do?page='+page+"&rows="+rows, searchEntity);
	}

    //商品审核
    this.updateStatus = function(ids,status){
        return $http.get('../goods/updateStatus.do?ids='+ids+"&status="+status);
    }

});